huaweicloud/services/fgs/resource_huaweicloud_fgs_dependency.go

```diff
@@ -2,162 +2,252 @@ package fgs
+		d.Set("version", utils.PathSearch("version", respBody, nil)),
 	)
 	if err := mErr.ErrorOrNil(); err != nil {
 		return diag.Errorf("error setting resource fields of custom dependency (%s): %s", d.Id(), err)
 	}
-
 	return nil
 }
 
-func resourceFgsDependencyUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	cfg := meta.(*config.Config)
-	client, err := cfg.FgsV2Client(cfg.GetRegion(d))
+func buildUpdateDependencyBodyParams(d *schema.ResourceData) map[string]interface{} {
+	return map[string]interface{}{
+		"name":        d.Get("name").(string),
+		"runtime":     d.Get("runtime").(string),
+		"description": d.Get("description").(string),
+		"depend_type": "obs",
+		"depend_link": d.Get("link").(string),
+	}
+}
+
+func resourceDependencyUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg          = meta.(*config.Config)
+		region       = cfg.GetRegion(d)
+		httpUrl      = "v2/{project_id}/fgs/dependencies/{depend_id}"
+		dependencyId = d.Id()
+	)
+
+	client, err := cfg.NewServiceClient("fgs", region)
 	if err != nil {
-		return diag.Errorf("error creating FunctionGraph v2 client: %s", err)
+		return diag.Errorf("error creating FunctionGraph client: %s", err)
+	}
+
+	updatePath := client.Endpoint + httpUrl
+	updatePath = strings.ReplaceAll(updatePath, "{project_id}", client.ProjectID)
+	updatePath = strings.ReplaceAll(updatePath, "{depend_id}", dependencyId)
+	updateOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders: map[string]string{
+			"Content-Type": "application/json",
+		},
+		JSONBody: utils.RemoveNil(buildUpdateDependencyBodyParams(d)),
 	}
 
-	opts := buildFgsDependencyOpts(d)
-	_, err = dependencies.Update(client, d.Id(), opts)
+	_, err = client.Request("PUT", updatePath, &updateOpt)
 	if err != nil {
 		return diag.Errorf("error updating custom dependency: %s", err)
 	}
-
-	return resourceFgsDependencyRead(ctx, d, meta)
+	return resourceDependencyRead(ctx, d, meta)
 }
 
-func resourceFgsDependencyDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	cfg := meta.(*config.Config)
-	fgsClient, err := cfg.FgsV2Client(cfg.GetRegion(d))
+func resourceDependencyDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg          = meta.(*config.Config)
+		region       = cfg.GetRegion(d)
+		httpUrl      = "v2/{project_id}/fgs/dependencies/{depend_id}"
+		dependencyId = d.Id()
+	)
+
+	client, err := cfg.NewServiceClient("fgs", region)
 	if err != nil {
-		return diag.Errorf("error creating FunctionGraph v2 client: %s", err)
+		return diag.Errorf("error creating FunctionGraph client: %s", err)
+	}
+
+	updatePath := client.Endpoint + httpUrl
+	updatePath = strings.ReplaceAll(updatePath, "{project_id}", client.ProjectID)
+	updatePath = strings.ReplaceAll(updatePath, "{depend_id}", dependencyId)
+	updateOpt := golangsdk.RequestOpts{
```

Comment:
How about changing variable name to `deletePath`, `deleteOpt`.

Reply:
done.